infrastructure/devicepathresolver/multipath_device_path_resolver.go

```diff
@@ -0,0 +1,21 @@
+package devicepathresolver
+
+import (
+	boshsettings "github.com/cloudfoundry/bosh-agent/settings"
+)
+
+type multipathDevicePathResolver struct {
+	usePreformattedPersistentDisk bool
+}
+
+func NewMultipathDevicePathResolver(usePreformattedPersistentDisk bool) DevicePathResolver {
+	return multipathDevicePathResolver{usePreformattedPersistentDisk: usePreformattedPersistentDisk}
+}
+
+func (r multipathDevicePathResolver) GetRealDevicePath(diskSettings boshsettings.DiskSettings) (string, bool, error) {
+	if r.usePreformattedPersistentDisk {
+		return diskSettings.Path, false, nil
+	} else {
+		return diskSettings.Path + "-part", false, nil
```

Comment:
If this is the only change to the Agent at this point, you could just make a change to softlayer CPI to return disk path including "-part". Then we would not have to have this PR.